In [53]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import math
# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory


# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

# NOMOR 1

In [32]:
pi=math.pi
def fungsi(h) :
    return pi*h**3-9*pi*h**2+90
    

In [31]:
def turunan(h) :
    return 3*pi*h**2-18*pi*h

In [33]:

def newton(x1) :
    x0=0
    x_1=[]
    x_0=[]
    
    while(abs(x1-x0)>0.00000001) :
       
        x0=x1
        x_0.append(x0)
        x1=x0-(fungsi(x0)/turunan(x0))
        x_1.append(x1)
    return x1,x_0,x_1
    

Kita inisiasi dengan x awal = 3 sebab jika x awal =0 , maka akan terjadi eror

In [34]:
x,x_0,x_1=newton(3)

In [37]:
df=pd.DataFrame()
df['h']=x_1
df['x0']=x_0

In [38]:
df

h        x0
0  2.061033  3.000000
1  2.027042  2.061033
2  2.026906  2.027042
3  2.026906  2.026906

Dapat dilihat bahwa iterasi terjadi sebanyak 4 kali dengan nilai akhir yaitu h pada 2.026906

# NOMOR 2

In [39]:
def function2(h) :
    return 15*h**2+5*h**3-16

In [40]:
def turunan2(h) :
    return 15*h**2+30*h

In [44]:

def newton2(x1) :
    x0=0
    x_1=[]
    x_0=[]
    
    while(abs(x1-x0)>0.00000001) :
       
        x0=x1
        x_0.append(x0)
        x1=x0-(function2(x0)/turunan2(x0))
        x_1.append(x1)
    return x1,x_0,x_1

Kita inisialisasi dengan x awal = 2

In [48]:
x2,x_2,x_2_0=newton2(2)

In [51]:
df2=pd.DataFrame()
df2['h']=x_2
df2['x0']=x_2_0

In [52]:
df2

h        x0
0  2.000000  1.300000
1  1.300000  0.983994
2  0.983994  0.909354
3  0.909354  0.905229
4  0.905229  0.905216
5  0.905216  0.905216

Dapat dilihat terjadi 6 kali iterasi dengan nilai akhir h= 0.905216